### PR TITLE
[inductor] Fix a use before def error

### PR DESCRIPTION
**Pull request by desertfire**
Summary: for https://github.com/pytorch/torchdynamo/issues/918

### CODE REVIEW

**Comment by desertfire** [comment]
There is still another error where `tl.load` fails for a dim=0 tensor, 
```
tmp4 =  tl.load(in_ptr1 + 0, tmp2)
...
arg1_1 = rand_strided((), (), device='cuda', dtype=torch.float32)
```

**Review by jansel — APPROVED** [review]

File: torchinductor/codegen/triton.py [hunk 1]
```diff
@@ -632,6 +638,7 @@ def indexing(self, index: sympy.Expr, copy_shape=None, dense_indexing=False):
             or indirect_indexing
             or self._load_mask is not None
         ) and index != 0
```
Comment (jansel) [inline torchinductor/codegen/triton.py:640]: The `index != 0` here seems related, I think this was related to loading the random mask.

File: torchinductor/codegen/triton.py [hunk 2]
```diff
@@ -646,7 +653,7 @@ def indexing(self, index: sympy.Expr, copy_shape=None, dense_indexing=False):
                 mask.append(f"{tree.prefix}mask")
             dense_mask.append(f"{tree.prefix}mask")
 
-        if need_dense and not have_dense:
+        if (need_dense and not have_dense) or load_index0:
```
Comment (jansel) [inline torchinductor/codegen/triton.py:656]: Why not
```suggestion
        if (need_dense and not have_dense) or index == 0:
```
?

Comment (desertfire) [reply torchinductor/codegen/triton.py]: I wasn't sure if this is safe to do if we are not calling from `load`, but CI run suggests this is ok.

**Comment by ngimel** [comment]
> There is still another error where `tl.load` fails for a dim=0 tensor,
> 

wdym? How does it fail?

**Comment by desertfire** [comment]
> > There is still another error where `tl.load` fails for a dim=0 tensor,
> 
> wdym? How does it fail?

The error I saw was,
```
>       name, asm, shared_mem = _triton.code_gen.compile_ttir(backend, generator.module, device, num_warps, num_stages, extern_libs)
E       MemoryError: std::bad_alloc

../triton/python/triton/code_gen.py:1320: MemoryError
```

Adding `tl.zeros([XBLOCK, RBLOCK]` to the indexing would make it pass, but as @ngimel pointed out in https://github.com/pytorch/torchdynamo/pull/981, what we really need here is mask as `None`.

**Pull request merged**